### PR TITLE
Integrate MLIR-EmitC at iml130/mlir-emitc@8a46b64b

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

**Review by GMNGeoffrey — APPROVED** [review]
Soon we're going to pull the remaining not-upstreamed stuff into IREE and not have to bump the submodule, right?

**Comment by marbre** [comment]
> Soon we're going to pull the remaining not-upstreamed stuff into IREE and not have to bump the submodule, right?

Well we still plan to upstream all stuff! However, this means that as we'll have to close the emitters API prior to upstreaming it. Whith this the upstreamed emitter can be used in IREE. For the time we'll need to enhance and refactor the EmitC dialect as well as the emitter, the plan is to pull in https://github.com/iml130/mlir-emitc/blob/main/lib/Target/Cpp/TranslateToCpp.cpp. Anyway, the final goal is to fully rely on the MLIR core repo and to get rid of the locally used emitter.

TL;DR: Yes, we won't have to bump the submodule any more in near future (probably next week?).
